app/controllers/NationalInsuranceNumberController.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions._
+import forms.YesNoPageFormProvider
+
+import javax.inject.Inject
+import models.{Mode, NameDOB}
+import models.SchemeId.Srn
+import navigation.Navigator
+import pages.{MemberDetailsPage, NationalInsuranceNumberPage}
+import play.api.data.Form
+import viewmodels.models.YesNoPageViewModel
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.YesNoPageView
+import NationalInsuranceNumberController._
+import models.requests.DataRequest
+import services.SaveService
+import viewmodels.DisplayMessage.SimpleMessage
+
+import scala.concurrent.{ExecutionContext, Future}
+import utils.FormUtils._
+
+class NationalInsuranceNumberController @Inject()(
+                                                   override val messagesApi: MessagesApi,
+                                                   saveService: SaveService,
+                                                   navigator: Navigator,
+                                                   identify: IdentifierAction,
+                                                   allowAccess: AllowAccessActionProvider,
+                                                   getData: DataRetrievalAction,
+                                                   requireData: DataRequiredAction,
+                                                   formProvider: YesNoPageFormProvider,
+                                                   val controllerComponents: MessagesControllerComponents,
+                                                   view: YesNoPageView
+                                                 )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+  private def form(memberName: String) = NationalInsuranceNumberController.form(formProvider, memberName)
+
+  def onPageLoad(srn: Srn, mode: Mode): Action[AnyContent] = (identify andThen allowAccess(srn) andThen getData andThen requireData).async {
```

Comment:
I added another the `IdentifyAndRequireData` action to shorten this `identifyAndRequireData(srn)`